### PR TITLE
✨ Add crossbrowser test

### PR DESCRIPTION
**Pull request by dominikkaminski**
**Changes:**

1. Add crossbrowser.com protractor configuration file
2. Move protractor.conf.js to test/protractor.conf.js
3. Add npm run crossbrowser-test command to start protractor test on crossbrowser.com
4. Add integration-test.sh inorder to set crossbrowser account details as environment variables
5. Add isLaunching test to check if BPMN-Studio is starting

## How can others test the changes?

```shell
npm run crossbrowser-test
```

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: aurelia_project/tasks/e2eCrossbrowser.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "name": "e2eCrossbrowser",
+    "description": "Runs all end-to-end (e2e) tests at crossbrowser.com and reports the results.",
+    "flags": []
+  }
+  
+
```
Comment (Vyperus) [inline aurelia_project/tasks/e2eCrossbrowser.json]: Unnecessary new line.

File: aurelia_project/tasks/e2eCrossbrowser.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/**
+ * e2e task
+ *
+ * You should have the server up and running before executing this task. e.g. run `au run`, otherwise the
+ * protractor calls will fail.
+ */
+import {CLIOptions} from 'aurelia-cli';
```
Comment (Vyperus) [inline aurelia_project/tasks/e2eCrossbrowser.ts]: Please sort this imports like this:

1) Frameworkstuff (aurelia-cli and such)
2) Thirdpartystuff (protractor and so on)
3) Our stuff (Everything that is inside the application like "./../../tsconfig.json")

This just serves better readability.

File: integration-test.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+export CB_USER=
+export CB_KEY=
+export aureliaUrl=
+
+# HOW-TO: source integration-test.sh
```
Comment (Vyperus) [inline integration-test.sh]: How to what exactly. Please use full comments like "Run this with... to..." if you need to comment this here.

Also missing new line.

File: test/protractor.conf.crossbrowser.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+exports.config = {
+  
+    // change this to your USERNAME and AUTHKEY
+    seleniumAddress: 'http://' + process.env.CB_USER +':'+ process.env.CB_KEY +'@hub.crossbrowsertesting.com:80/wd/hub',
+  
+    capabilities : {
+  
+      name : 'BPMN Studio test', // this will show up in the UI
+      
+      // these are important :)
```
Comment (Vyperus) [inline test/protractor.conf.crossbrowser.js]: Well sure they are important but if so what do they do? This comment doesn't give anyone a clue about it.
If you feel the need of comment here please describe it in more detail ;)

**Comment by dominikkaminski** [comment]
Die gewünschten Änderungen sind eingebaut @Vyperus

File: .gitignore [hunk 5]
```diff
@@ -35,3 +35,6 @@ bootstrap
 
 process-engine_local_db
 userData
+
+# Personal integration-test settings
+integration-test.sh
```
Comment (Vyperus) [inline .gitignore]: Missing new line

File: aurelia_project/tasks/e2eCrossbrowser.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+    "name": "e2eCrossbrowser",
+    "description": "Runs all end-to-end (e2e) tests at crossbrowser.com and reports the results.",
+    "flags": []
+  }
```
Comment (Vyperus) [inline aurelia_project/tasks/e2eCrossbrowser.json]: Wrong indent in this file.

File: .gitignore (hunk 5 above)
Comment (cmg-dev) [reply .gitignore]: Missing newline at the end of file.

@dominikkaminski This is important for all tool to work.
And important for your teamleader ;)

And even Github is complaining about this ;)

File: aurelia_project/tasks/e2eCrossbrowser.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+import * as del from 'del';
+import * as gulp from 'gulp';
+import {protractor} from 'gulp-protractor';
+import * as typescript from 'gulp-typescript';
+
+import * as tsConfig from './../../tsconfig.json';
+import * as project from './../aurelia.json';
```
Comment (cmg-dev) [inline aurelia_project/tasks/e2eCrossbrowser.ts]: Sorting the imports is a rather important step in our quality control.

You will learn this by review, or have a look around in other code bases :)

Comment (cmg-dev) [reply aurelia_project/tasks/e2eCrossbrowser.ts]: Sidenote: We do this in ALL technologies and languages.

File: integration-test.sh [hunk 8]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+export CB_USER=
+export CB_KEY=
+export aureliaUrl=
```
Comment (cmg-dev) [inline integration-test.sh]: Do they have to be empty? 

Wouldn't it be better to:

```bash
export CB_USER=""
````
?

File: test/e2e/src/isLaunching.spec.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import {browser, by, element} from 'protractor';
+
+describe('isLaunching', () => {
```
Comment (cmg-dev) [inline test/e2e/src/isLaunching.spec.ts]: okay we should maybe call this:

```js
describe('basic_layout_tests', () => {
(...)
```

File: test/e2e/src/isLaunching.spec.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+  browser.driver.manage().deleteAllCookies();
+  browser.get(aureliaUrl);
+
+  it('should display the satusbar', () => {
+    browser.get(aureliaUrl);
+    browser.sleep(defaultTimeoutMS);
+    expect(element.all(by.tagName('status-bar')).count()).toBeGreaterThan(0);
+  });
+});
```
Comment (cmg-dev) [inline test/e2e/src/isLaunching.spec.ts]: Do we gain a benefit by using TypeScript for our Browser tests?

Please elaborate :)

File: aurelia_project/tasks/e2eCrossbrowser.ts (hunk 2 above)
Comment (dominikkaminski) [reply aurelia_project/tasks/e2eCrossbrowser.ts]: Done 👌

File: .gitignore (hunk 5 above)
Comment (dominikkaminski) [reply .gitignore]: Done 👌

File: aurelia_project/tasks/e2eCrossbrowser.ts (hunk 7 above)
Comment (dominikkaminski) [reply aurelia_project/tasks/e2eCrossbrowser.ts]: Done 👌

File: test/e2e/src/isLaunching.spec.ts (hunk 9 above)
Comment (dominikkaminski) [reply test/e2e/src/isLaunching.spec.ts]: Done 👌

File: test/e2e/src/isLaunching.spec.ts (hunk 10 above)
Comment (dominikkaminski) [reply test/e2e/src/isLaunching.spec.ts]: Elaborated 👍

**Comment by cmg-dev** [comment]
@dominikkaminski Thanks, please merge.

**Pull request merged**